src/app/inventory/store/d2-item-factory.ts

```diff
@@ -76,23 +78,27 @@ export function processItems(
   owner: DimStore,
   items: DestinyItemComponent[],
   itemComponents: DestinyItemComponentSetOfint64,
-  uninstancedItemObjectives?: {
-    [key: number]: DestinyObjectiveProgress[];
-  },
-  profileRecords?: DestinyProfileRecordsComponent
+  uninstancedItemObjectives?: DestinyCharacterProgressionComponent['uninstancedItemObjectives'],
+  profileRecords?: DestinyProfileRecordsComponent,
+  uninstancedItemPerks?: DestinyCharacterProgressionComponent['uninstancedItemPerks']
```

Comment:
these calculate out to the same type when you hover or use the param, but more explicitly show how we expect to get the information from the characters' progressions component